client/src/components/types.ts

```diff
@@ -0,0 +1,24 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+export interface ChatMsg {
+  senderId: string;
+  type: string;
```

Comment:
If were me, I would change `type` here to an enum, e.g.

```typescript
export enum ChatMsgType {
  text: 'text',
  feedbackPositive: 'feedbackPositive',
 ...
}

export interface ChatMsg {
  ...
  type: ChatMsgType
}
```

This one isn't required, but I do think it's better

Reply:
Only thing is that these keep changing

Reply:
yes. that's why I prefer to have them typed and living in one file. That way when you change them,  the compiler will find for you all the places in {SomeComponent}.tsx where there are string literals for types that might be wrong

Reply:
*Facepalm* I'm so silly. Making the change rn